### PR TITLE
Game Lab: Import Kenney Jumper Assets

### PR DESCRIPTION
**Pull request by islemaster**
out

Updates the animation library manifest to add updated assets from the [Kenney.nl Jumper Pack](http://kenney.nl/assets/jumper-pack), bringing our library to 111 animations.  Metadata generated using a new script (included here) `generateSingleFrameAnimationMetadata.rb` that will spit out metadata JSON files for a set of single-frame PNGs.  It's used like this:

```
> tools/scripts/generateSingleFrameAnimationMetadata.rb ~/Downloads/jumper/PNG/Players/bunny2*.png --aliases jumper,player,purple
```

There are a number of these assets that we'll want to go back and merge into proper animations, and more that we'll probably want to hand-craft better aliases for.  That will be a good test of the library update functionality, so I want to go ahead and get these in sooner rather than later.

Note, there are also still TODOs in the manifest rebuilding script - those are being dealt with soon to make the script more friendly for PM use later, but are out of scope for this PR.

### CODE REVIEW

**Comment by islemaster** [comment]
Note to @joshlory: You can mostly skip the huge JSON file, which is autogenerated.

File: lib/cdo/png_utils.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  # Given the complete body of a PNG file, returns hash of source dimensions:
+  # {"x": _, "y": _}
+  def dimensions_from_png(png_body)
+    # Read the first eight bytes of the IHDR Chunk, which must always be the
+    # first chunk of the PNG file.
+    #
+    # PNG Header takes 8 bytes (0x00-0x07)
+    # IHDR chunk length takes 4 bytes (0x08-0x0b)
+    # IHDR chunk type code takes 4 bytes (0x0c-0x0f)
+    # IHDR chunk data begins at 0x10
```
Comment (joshlory) [inline lib/cdo/png_utils.rb:14]: This is clever, but I wonder if it's better to not roll our own?  We use MiniMagick elsewhere:

``` ruby
MiniMagick::Image.open(filename).dimensions
=> [2000, 1500]
```

Comment (islemaster) [reply lib/cdo/png_utils.rb:14]: Ah, didn't know what look look for on this.  I'll change it over.

**Comment by islemaster** [comment]
We never upload `animationLibrary.json` to S3 - by itself, anyway. It gets checked into git and used by the apps build, where it becomes part of the gamelab bundle.  That bundle goes to S3 (both minified and unminified versions).

I like checking in the unminified JSON because we'll get nice diffs for small changes to library image metadata. It's less useful for large content dumps like this though.

We're also not minifying the individual animation metadata files that we do push to S3 because they are source material, which we want to keep editable.  Also, they're generally under 200 bytes each, anyway.

**Comment by joshlory** [comment]
Sounds good.  LGTM!

**Pull request merged**